### ISSUE TITLE
APSFOUND-22: REST-based test connection service

### ISSUE THREAD

**Issue by njr-11** [OpenLiberty/open-liberty#6509]
Rebase the prototype test connection service onto OpenAPI, move it to Open Liberty, and release as GA function.

The following blog post describes the latest beta feature that covers validation for JDBC, JCA, JMS and cloudant database.   It does not support API Discovery,
https://openliberty.io/blog/2019/05/24/testing-database-connections-REST-APIs.html

This needs to be rebased onto OpenAPI.

**Comment by marikaj123** [comment]
Andy has confirmed the beta code is in a green release build.  Tagging it for beta 19005.

**Comment by skasund** [comment]
L2 has requested STE slides for this feature.  The STE template can be found at the link below: 

https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm

Please upload the completed slides to the same BOX folder.  Thanks!

**Comment by chirp1** [comment]
No ID required. Approving.

**Comment by marikaj123** [comment]
FYI: After the appropriate approvals are granted then the Epic kind=GA can become enabled.

**Comment by marikaj123** [comment]
According to the feature owner, the returned translated files exist in Master.

**Comment by dave-waddling** [comment]
**FAT Focal Approval comment:** Hi, please can you complete the Feature Test Summary issue (#8656) which I have added to this epic - it is used to review the FAT testing and obtain the FAT Focal Approval.

**Comment by dave-waddling** [comment]
**FAT Focal Approval comment:** Hi, I've reviewed the Feature Test Summary (#8656) and all is looking good there. That said, defect 266297 (which looks to be a Windows specific issue) needs investigation/fixing before approval can be granted - it's occurred in the two latest SOEs and our standard Windows platform build.

**Comment by dave-waddling** [comment]
**FAT Focal Approval comment:** Hi @njr-11, thanks for the update. None of the builds I reviewed contained #8689 - I was looking at the SOE builds from last week as all of your code (apart from that fix) was in them. As this weekend's SOE will run against a Release build containing #8689 I'm optimistic that all of your testing will pass cleanly and I'll be able to grant a FAT Focal Approval on Tuesday (next Monday is a Bank Holiday here in the UK).

**Comment by dave-waddling** [comment]
**FAT Focal Approval comment:** Hi, as per my previous comment the testing for this feature all passed cleanly in the SOE and so I'm adding the FAT focal approval at this time.

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

**Comment by skasund** [comment]
Providing STE approval early.  Please provide the slides as soon as possible.  Thanks!
